### PR TITLE
lint: Add tests/infrastructure to linter

### PR DESCRIPTION
**Pull request by brianmcarey**
### What this PR does

Add tests/infrastructure to linter and fix resulting lint issues. 

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/infrastructure/certificates.go [hunk 1]
```diff
@@ -117,30 +120,34 @@ var _ = DescribeInfra("[rfe_id:4102][crit:medium][vendor:cnv-qe@redhat.com][leve
...
+			apiService, err := aggregatorClient.ApiregistrationV1().APIServices().Get(
+				context.Background(), fmt.Sprintf("%s.subresources.kubevirt.io", v1.ApiLatestVersion), metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 			return libinfra.ContainsCrt(apiService.Spec.CABundle, newCA)
 		}, 10*time.Second, 1*time.Second).Should(BeTrue())
 
 		By("checking that we can still start virtual machines and connect to the VMI")
 		vmi := libvmifact.NewAlpine()
-		vmi = libvmops.RunVMIAndExpectLaunch(vmi, 60)
+		vmiStartTimeOut := 60
```
Comment (0xFelix) [inline tests/infrastructure/certificates.go:131]: nit: Could be a const instead.

File: tests/infrastructure/certificates.go [hunk 2]
```diff
@@ -117,30 +120,34 @@ var _ = DescribeInfra("[rfe_id:4102][crit:medium][vendor:cnv-qe@redhat.com][leve
...
-		err = virtClient.CoreV1().Secrets(flags.KubeVirtInstallNamespace).Delete(context.Background(), components.KubeVirtCASecretName, metav1.DeleteOptions{})
+		err := virtClient.CoreV1().Secrets(flags.KubeVirtInstallNamespace).Delete(
+			context.Background(), components.KubeVirtCASecretName, metav1.DeleteOptions{})
 		Expect(err).ToNot(HaveOccurred())
 
 		By("repeatedly starting VMIs until virt-api and virt-handler certificates are updated")
 		Eventually(func() (rotated bool) {
 			vmi := libvmifact.NewAlpine()
-			vmi = libvmops.RunVMIAndExpectLaunch(vmi, 60)
+			vmiStartTimeout := 60
```
Comment (0xFelix) [inline tests/infrastructure/certificates.go:149]: Same

File: tests/infrastructure/prometheus.go [hunk 3]
```diff
@@ -617,22 +641,16 @@ func countReadyAndLeaderPods(pod *k8sv1.Pod, component string) (foundMetrics map
 		}
 
 		cmd := fmt.Sprintf("curl -L -k https://%s:8443/metrics", libnet.FormatIPForURL(ep.IP))
-		stdout, stderr, err := exec.ExecuteCommandOnPodWithResults(pod, "virt-handler", strings.Fields(cmd))
+		var stdout, stderr string
+		stdout, stderr, err = exec.ExecuteCommandOnPodWithResults(pod, "virt-handler", strings.Fields(cmd))
```
[the remark this answers is not in the document]
Comment (brianmcarey) [reply tests/infrastructure/prometheus.go:645]: It was giving out about shadowing the err above in L629

https://github.com/kubevirt/kubevirt/pull/13274/files/177d1f43af3f21ca96b5d88424e7a46e93942fef#diff-9ad6a6110d7d4105a50147793d748edfaa2237b821bd4e5a6f512b5cb3030ef7R629

```
tests/infrastructure/prometheus.go:620:19: shadow: declaration of "err" shadows declaration at line 603 (govet)
		stdout, stderr, err := exec.ExecuteCommandOnPodWithResults(pod, "virt-handler", strings.Fields(cmd))
```

I wasn't a fan of this one either but its not the worst either.

File: tests/infrastructure/prometheus.go [hunk 4]
```diff
@@ -617,22 +641,16 @@ func countReadyAndLeaderPods(pod *k8sv1.Pod, component string) (foundMetrics map
...
-			case fmt.Sprintf("kubevirt_virt_%s_leading_status 1", component):
-				foundMetrics["leading"]++
-			case fmt.Sprintf("kubevirt_virt_%s_ready_status 1", component):
-				foundMetrics["ready"]++
-			}
+		if strings.Contains(stdout, leadingMetric) {
+			foundMetrics["leading"]++
+		}
+		if strings.Contains(stdout, readyMetric) {
+			foundMetrics["ready"]++
```
[the remark this answers is not in the document]
Comment (brianmcarey) [reply tests/infrastructure/prometheus.go:653]: I'll rework this.

File: tests/infrastructure/virt-handler.go [hunk 5]
```diff
@@ -76,8 +76,10 @@ var _ = DescribeInfra("virt-handler", func() {
 
 	restoreNodes := func(nodes []string) {
 		for _, node := range nodes {
-			patchBytes := []byte(fmt.Sprintf(`[{"op": "remove", "path": "/metadata/annotations/%s"}, {"op": "remove", "path": "/metadata/annotations/%s"}]`,
-				strings.ReplaceAll(v1.KSMFreePercentOverride, "/", "~1"), strings.ReplaceAll(v1.KSMPagesDecayOverride, "/", "~1")))
+			patchBytes := []byte(fmt.Sprintf(
```
[the remark this answers is not in the document]
Comment (brianmcarey) [reply tests/infrastructure/virt-handler.go:79]: +1

File: tests/infrastructure/certificates.go (hunk 1 above)
Comment (brianmcarey) [reply tests/infrastructure/certificates.go:131]: +1

**Comment by brianmcarey** [comment]
@0xFelix thanks for the review - I've addressed your comments - can you take a look again when you get a chance?

File: tests/infrastructure/taints-and-tolerations.go [hunk 6]
```diff
@@ -227,12 +234,13 @@ func getDeploymentsForPods(virtClient kubecli.KubevirtClient, pods []k8sv1.Pod)
...
 	var result []types.NamespacedName
-	for _, deployment := range allDeployments.Items {
+	for i := range allDeployments.Items {
+		deployment := allDeployments.Items[i]
 		selector, err := metav1.LabelSelectorAsSelector(deployment.Spec.Selector)
 		Expect(err).NotTo(HaveOccurred())
 
-		for _, pod := range pods {
-			if selector.Matches(labels.Set(pod.Labels)) {
+		for y := range pods {
```
Comment (dhiller) [inline tests/infrastructure/taints-and-tolerations.go:242]: nit: for loop index variables are named in general like this: {'i', 'k', 'l', ..} - was just odd to me when reading though, will not block on this.

File: tests/infrastructure/virt-handler.go [hunk 7]
```diff
@@ -42,12 +43,12 @@ import (
 )
 
 var _ = DescribeInfra("virt-handler", func() {
-
 	var (
 		virtClient       kubecli.KubevirtClient
 		originalKubeVirt *v1.KubeVirt
 		nodesToEnableKSM []string
 	)
+	const trueStr = "true"
```
Comment (dhiller) [inline tests/infrastructure/virt-handler.go:51]: there it is, the other occurrence.

File: tests/infrastructure/node-labeller.go [hunk 8]
```diff
@@ -50,10 +50,10 @@ import (
 )
 
 var _ = DescribeInfra("Node-labeller", func() {
+	const trueStr = "true"
```
[the remark this answers is not in the document]
Comment (brianmcarey) [reply tests/infrastructure/node-labeller.go:53]: Trying this suggestion leads to an error 

`strconv.FormatBool(true) (value of type string) is not a constant`

so is it ok to leave it as is?

File: tests/infrastructure/taints-and-tolerations.go (hunk 6 above)
Comment (brianmcarey) [reply tests/infrastructure/taints-and-tolerations.go]: Addressed this in the latest push

File: tests/infrastructure/node-labeller.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (dhiller) [reply tests/infrastructure/node-labeller.go:53]: OK then... :smiling_face_with_tear:

**Review by dhiller — APPROVED** [review]
/hold

to give Felix a chance to review the latest version.

File: tests/infrastructure/node-labeller.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (0xFelix) [reply tests/infrastructure/node-labeller.go:53]: Could be `var trueStr = strconv.FormatBool(true)` though. :sweat_smile:

**Review by 0xFelix — APPROVED** [review]
/lgtm

Looks fine to me other than the nits about `trueStr`.

File: tests/infrastructure/node-labeller.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (brianmcarey) [reply tests/infrastructure/node-labeller.go:53]: I will take that as a quick followup - I think its worth getting these files covered by the linter so I am going to remove the hold now.

**Comment by brianmcarey** [comment]
@dhiller rebased - can you take another look if you get a chance?

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Review by dhiller — APPROVED** [review]

**Pull request merged**